src/main/scala/viper/gobra/reporting/Source.scala

```diff
@@ -17,6 +17,9 @@ object Source {
 
   case class Origin(pos: SourcePosition, tag: String)
 
+  abstract class Annotation()
+  class AnnotatedOrigin(val origin: Origin, val annotation: Annotation) extends Origin(origin.pos, origin.tag)
```

Comment:
could this be a case class?

Reply:
I don't think it can because it extends `Origin` which is itself a case class and case-to-case inheritance is prohibited. I could however create an `AbstractOrigin` class that is extended by both `Origin` and `AnnotatedOrigin`. Would this be a better option?

Reply:
yes